third_party/libusb/naclport/src/chrome_usb/chrome-usb-backend.js

```diff
@@ -228,10 +231,43 @@ ChromeUsbBackend.prototype.reportRequestError_ = function(
  */
 ChromeUsbBackend.prototype.reportRequestSuccess_ = function(
     debugRepresentation, promiseResolver, resultArgs) {
+  if (this.fakeNoDevices &&
+      (debugRepresentation.startsWith('chrome.usb.getDevices') ||
+      debugRepresentation.startsWith('chrome.usb.getConfigurations'))) {
+    resultArgs[0] = [];
+  }
   this.logger.fine(
       'Results returned by the ' + debugRepresentation + ' call: ' +
       goog.iter.join(goog.iter.map(resultArgs, debugDump), ', '));
   promiseResolver.resolve(resultArgs);
 };
 
+/**
+ * @param {!chrome.loginState.ProfileType} profileType
+ * @private
+ */
+ChromeUsbBackend.prototype.receiveProfileType_ = function(profileType) {
+  if (chrome.loginState &&
+      profileType === chrome.loginState.ProfileType.USER_PROFILE) {
+    chrome.loginState.onSessionStateChanged.addListener(
```

Comment:
It might be that the session state was already in the undesired state by this point. I think there's no other way than also using getSessionState() in order to handle this correctly.

As a side note, it would be great if we could have unit test coverage for these scenarios.